lib/command.js

```diff
@@ -26,15 +26,15 @@ module.exports = function command (yargs, usage, validation, globalMiddleware) {
     if (Array.isArray(cmd)) {
       aliases = cmd.slice(1)
       cmd = cmd[0]
-    } else if (typeof cmd === 'object') {
+    } else if (typeof cmd === 'object' || typeof cmd === 'function') {
```

Comment:
let's pull this into a helper that insures, if `cmd` is a `function`, it has appropriate fields, e.g.,

It should have a `description`, or `builder`, or `handler` or `middlewares`.

Perhaps we could apply the same check for when `typeof cmd === 'object'`, and simply do:

```
else if (isCommandDescription(cmd)) {
      let command = (Array.isArray(cmd.command) || typeof cmd.command === 'string') ? cmd.command : moduleName(cmd)
      if (cmd.aliases) command = [].concat(command).concat(cmd.aliases)
      self.addHandler(command, extractDesc(cmd), cmd.builder, cmd.handler, cmd.middlewares)
}
```